ReactiveCocoa/Swift/Signal.swift

```diff
@@ -1031,6 +1031,40 @@ extension SignalType {
 	}
 }
 
+extension SignalType where Value: Hashable {
+	/// Forwards only those values from `self` that are unique across the set of
+	/// all values that have been seen
+	/// Note: This causes values that are forwarded to be retained to check for
+	/// uniquness
+	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
+	public func uniqueValues() -> Signal<Value, Error> {
+		return Signal { observer in
+			var set = Set<Value>()
```

Comment:
Using `NSLock` is fine, but I think it would be simpler to use `Atomic<Set<Value>>` instead of implementing an adhoc locking mechanism here. Then you can use `Atomic.modify` to `insert` and `Atomic.withValue` to check if it `contains` it :)